### ISSUE TITLE
Q: How to use in CI

### ISSUE THREAD

**Issue by thim81** [tebelorg/TagUI#29]
Hi Ken,

Love the simplicity of the TagUI and all the features that come with it.

I was hoping to plug it in our CI flow, but it's not clear how to use the report option or is there an option to output the report as "test coverage"?

Perhaps you have some tips or pointers how to use TagUI in a Continuous integration flow?

**Comment by thim81** [comment]
I think I found it, by adding the `test` when running the `./tagui flow_filename firefox test`it does all the testing and generates a proper test report.

**Comment by kensoh** [comment]
Thanks @thim81 for your feedback and raising this question. Admittedly test automation is not TagUI's strength. While it can be used for test automation, in the developers reference section, I recommended this list of [test automation tools](https://github.com/atinfo/awesome-test-automation) which may be a better fit.

That said, by running TagUI with the test option you can leverage on the test assertions and testing framework of CasperJS. An XUnit XML file will be created which you can integrate with CI tools. There are more details on this in the [testing section](https://github.com/tebelorg/TagUI#testing) of developers reference. Below is a sample. We can leave this issue open to see any more inputs from you or other users. Lol we were typing at the same time.

**TagUI flow file popup_test**
```
http://tebel.org/index_mobile.php
test.assertTextExists('ABOUT','Check for ABOUT text');
click Open-source software
popup open-source_software
{
test.assertSelectorHasText(tx('gist-data'),'Creative Commons','Check for key words in gist-data element');
snap page
}
```

**Output of tagui popup_test test**
```
TA:Desktop kensoh$ tagui popup_test test
Test file: /Users/kensoh/Desktop/popup_test.js                                  
Wed Jul 26 2017 23:57:48 GMT+0800 (+08)
# /Users/kensoh/Desktop/popup_test
# http://tebel.org/index_mobile.php - Tebel.Automation
PASS Check for ABOUT text
# click Open-source software
# popup open-source_software
PASS Check for key words in gist-data element
# snap page
# http://tebel.org/index_mobile.php - Tebel.Automation
PASS /Users/kensoh/Desktop/popup_test (2 tests)
PASS 2 tests executed in 3.775s, 2 passed, 0 failed, 0 dubious, 0 skipped.      
Result log stored in /Users/kensoh/Desktop/popup_test.xml
```

**Generated popup_test.xml**
```xml
This XML file does not appear to have any style information associated with it. The document tree is shown below.
<testsuites time="3.775">
<testsuite name="/Users/kensoh/Desktop/popup_test" tests="2" failures="0" errors="0" time="3.775" timestamp="2017-07-26T15:57:52.463Z" package="Users/kensoh/Desktop/popup_test">
<testcase name="Check for ABOUT text" classname="Users/kensoh/Desktop/popup_test" time="0.793"/>
<testcase name="Check for key words in gist-data element" classname="Users/kensoh/Desktop/popup_test" time="2.982"/>
<system-out>
<![CDATA[ ]]>
</system-out>
</testsuite>
</testsuites>
```

**Comment by thim81** [comment]
@kensoh 

I like the clean syntax for writing the test and it does exactly what I needed, so I'm going to try to setup TagUI as test suite in our CI workflow.

If I got it working, I'll post back to share how I did it with other users.

**Comment by kensoh** [comment]
Adding some notes related to CI. There is recent appearance of test / browser automation projects base on headless Chrome. I made a list here - [Chromeless, Chrominator, Chromy, Navalia, Lambdium, GhostJS, AutoGCD](https://medium.com/@kensoh/chromeless-chrominator-chromy-navalia-lambdium-ghostjs-autogcd-ef34bcd26907) with short notes.

Also, I've made a commit on the awesome test automation list recommended in developers reference to include these up and coming tools :smile: - https://github.com/atinfo/awesome-test-automation/blob/master/javascript-test-automation.md#web-ui-test-automation
